nifi/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteOnAttribute.java

```diff
@@ -166,6 +170,27 @@ public void onPropertyModified(final PropertyDescriptor descriptor, final String
         this.relationships.set(newRelationships);
     }
 
+    /**
+     * When this processor is 
+     * @param context
+     */
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        for (final PropertyDescriptor descriptor : context.getProperties().keySet()) {
+            if (!descriptor.isDynamic()) {
+                continue;
+            }
+            getLogger().debug("Adding new dynamic property: {}", new Object[]{descriptor});
+            propertyMap.put(new Relationship.Builder().name(descriptor.getName()).build(), context.getProperty(descriptor));
+        }
+    }
+    
+    @OnUnscheduled
```

Comment:
Can we changed this to @OnStopped? As it is right now, we have a race condition - map could get cleared just after a thread enters onTrigger, which would result in it running with an empty map